website/blog/2022-04-28-build-open-lakehouse-using-apache-hudi-and-dbt.md

```diff
@@ -0,0 +1,180 @@
+
+To use incremental models, you need to perform these two activities:
+1. Tell dbt how to filter the rows on the incremental executions
+2. Define the uniqueness constraint of the model (required when using >= Hudi 0.10.1 version)
+
+### How to apply filters on an incremental run?
+
+dbt provides you a macro `is_incremental()` which is very useful to define the filters exclusively for incremental materializations.
+
+Often, you'll want to filter for "new" rows, as in, rows that have been created since the last time dbt ran this model. The best way to find the timestamp of the most recent run of this model is by checking the most recent timestamp in your target table. dbt makes it easy to query your target table by using the "[{{ this }}](https://docs.getdbt.com/reference/dbt-jinja-functions/this)" variable.
+
+```sql title="models/my_model.sql"
+{{
+   config(
+       materialized='incremental',
+       file_format='hudi',
+   )
+}}
+
+select
+   *
+from raw_app_data.events
+{% if is_incremental() %}
+   -- this filter will only be applied on an incremental run
+   where event_time > (select max(event_time) from {{ this }})
+{% endif %}
+```
+
+### How to define the uniqueness constraint?
+
+A unique_key is the primary key of the dataset, which determines whether a record has new values and should be updated/deleted, or inserted.
+
+You can define the unique_key in the configuration block at the top of your model. This unique_key will act as the primaryKey (hoodie.datasource.write.recordkey.field) on the hudi table.
+
+## Step 4: How to use the upsert feature while writing datasets?
+
+dbt offers multiple load strategies when loading the transformed datasets, such as:
+* append (default)
+* insert_overwrite (optional)
+* merge (optional, Only available for Hudi and Delta formats)
+
+By default dbt uses the append strategy, which may cause duplicate rows when you execute dbt run command multiple times on the same payload.
+
+When you choose the insert_overwrite strategy, dbt will overwrite the entire partition or full table load for every dbt run, which causes unnecessary overheads and is very expensive.
+
+In addition to all the existing strategies to load the data, with hudi you can use the exclusive merge strategy when using incremental materialization. Using the merge strategy you can perform field-level updates/deletes on your data lakehouse which is performant and cost-efficient. As a result, you will get access to fresher data and accelerated insights.
+
+### How to perform field-level updates?
+
+If you are using the merge strategy and have specified a unique_key, by default, dbt will entirely overwrite matched rows with new values.
+
+Since Apache Spark adapter supports the merge strategy, you may optionally pass a list of column names to a merge_update_columns config. In that case, dbt will update _only_ the columns specified by the config, and keep the previous values of other columns.
```

Comment:
merge_update_columns config is not getting printed properly in the screenshot attached by you. Similarly update_only is coming with a underscore in between. 

Please check this line, rest looks good to me

Reply:
Good catch, the markdown syntax was wrong, I fixed the issue you flagged, thanks!